pkg/scheduler/preemption/preemption.go

```diff
@@ -141,6 +141,13 @@ func (p *Preemptor) issuePreemptions(ctx context.Context, targets []*workload.In
 			errCh.SendErrorWithCancel(err, cancel)
 			return
 		}
+
+		err = workload.UpdateStatus(ctx, p.client, target.Obj, kueue.WorkloadEvicted, metav1.ConditionTrue, kueue.WorkloadEvictedByPreemption, "Preempted to accommodate a higher priority Workload", "evict")
```

Comment:
Shouldn't `applyPreemption` change to add the Evicted condition only?

Also, we should avoid separate API calls. Consider that the apiserver could accept the first API call, but somehow fail the 2nd one. Or maybe the scheduler could immediately readmit the workload.

Reply:
Let me know once #692 is ready for review. I think I want to see the whole picture before merging this PR.

Reply:
I'm keeping #692 in draft jut to wait for this to be merged, you can have a look at any point.